libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -638,10 +655,10 @@ bool AP_Mount_Siyi::record_video(bool start_recording)
 
     // check desired recording state has changed
     bool ret = true;
-    if (_last_record_video != start_recording) {
+    const RecordingStatus new_record_status = start_recording ? RecordingStatus::ON : RecordingStatus::OFF;
+    if (_config_info.record_status != new_record_status) {
```

Comment:
I wonder if this is correct in cases where the camera has reported that its record_status is NO_CARD or DATA_LOSS.  I think that NO_CARD is essentially the same as OFF and maybe DATA_LOSS is the same as ON?

It matters because the silly API only allows toggling the start/stop.

Reply:
You're right. I would assume that both `DATA_LOSS` and `NO_CARD` are equivalent to `OFF`. 

I've updated to capture this.